version: '3'

networks:
  local:

services:

  checkout_service:
    container_name: checkout_service_1
    build:
      context: .
      dockerfile: Dockerfile
    environment:
      - ENV=local
      - DEBUG=True
      - PYTHONPATH=/checkouts:$PYTHONPATH
      - SECRET_KEY=RJGbf7r9nBbkANvadEdX4{4VJR
      - DATABASE_NAME=cabify_checkouts.db
      - FLASK_APP=server.py
    working_dir: /checkouts/
    volumes:
    - "$BACKEND_SERVICE_CODE/checkouts:/checkouts/"
    entrypoint:
      - /bin/sh
      - -c
      - |
          /usr/local/bin/gunicorn --workers "2" --log-level debug --error-logfile /tmp/gunicorn_err.log --reload --bind 0.0.0.0:8000 server:app
    ports:
      - "8005:8000"
    networks:
      local:
        aliases:
          - checkout_service
